x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/StackTrace.java

```diff
@@ -215,6 +215,17 @@ public static StackTrace fromSource(Map<String, Object> source) {
         return new StackTrace(addressOrLines, fileIDs, frameIDs, typeIDs, 0, 0, 0);
     }
 
+    public void forNativeAndKernelFrames(java.util.function.Consumer<String> consumer) {
+        final int NATIVE_FRAME_TYPE = 3;
```

Comment:
Can we turn these into class constants?

Reply:
My opinion on this is: As long as these values are just used in a single place, I normally keep them close to the code. That increases readability as you don't have to scroll or search for usages.
Creating class constants and widening the scope of access makes sense when used in 2+ places.

But I'll change it to not get lost in basic discussions 😃